test_conformance/subgroups/test_ifp.cpp

```diff
@@ -302,16 +302,27 @@ static test_status checkIFPSupport(cl_device_id device, bool &ifpSupport)
 {
     cl_uint ifp_supported;
     cl_uint error;
-    error = clGetDeviceInfo(device,
-                            CL_DEVICE_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS,
-                            sizeof(ifp_supported), &ifp_supported, NULL);
-    if (error != CL_SUCCESS)
+    auto device_cl_version = get_device_cl_version(device);
+    if (device_cl_version < Version(2, 1))
     {
-        print_error(
-            error,
-            "Unable to get CL_DEVICE_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS "
-            "capability");
-        return TEST_FAIL;
+        // OpenCL 2.0 and below don't provide the CL_DEVICE_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS property. Instead,
+        // support is determined by the presence of cl_khr_subgroups.
+        ifp_supported = is_extension_available(device, "cl_khr_subgroups");
```

Comment:
This duplicates the check in test_ifp_ext, which is used to determine if test should skip itself. Can we move that logic in here as well?

Reply:
I went with the approach of moving the check to test_ifp_ext as recommended by @gwawiork because:
- This function gets called from test_ifp_core as well, where it appears we only want to check for CL_DEVICE_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS (since this is part of core?, anyway that is the logic of the current code)
- When running test_ifp_ext, if version < 2.1 we only check for the string, if >= 2.1 we check for string and CL_DEVICE_SUB_GROUP_INDEPENDENT_FORWARD_PROGRESS.